### PR TITLE
Jan10

### PR DESCRIPTION
**Pull request by muhammadanas123**

### CODE REVIEW

File: spec/models/journey_spec.rb [hunk 1]
```diff
@@ -0,0 +1,68 @@
+require 'rails_helper'
+
+RSpec.describe Journey, type: :model do
+  context "when creating a journey" do
+
+    it "should be valid journey with all attributes" do
+      traveller = Traveller.create(firstname: "anas",
```
Comment (momnanaseer) [inline spec/models/journey_spec.rb:7]: use factory for data creation in test cases

File: spec/models/journey_spec.rb [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      expect(journey.valid?).to eq(true)
+    end
+
+    it "should validates the from" do
+      journey = Journey.new(from: "lahore")
+      expect(journey).to be_valid
+    end
+
+    it "should validates the lastname" do
+      journey = Journey.new(to: "islamabad")
```
Comment (momnanaseer) [inline spec/models/journey_spec.rb:32]: please save this and then expect true case

File: spec/models/journey_spec.rb [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+      expect(journey).to be_valid
+    end
+
+    it "should validates the departure date and arrival date" do
+      journey = Journey.new(departure_date: "22-10-2022", arrival_date: "23-11-2022")
```
Comment (momnanaseer) [inline spec/models/journey_spec.rb:37]: save record and expect error message

File: spec/models/journey_spec.rb (hunk 1 above)
Comment (muhammadanas123) [reply spec/models/journey_spec.rb]: done

File: spec/models/journey_spec.rb (hunk 2 above)
Comment (muhammadanas123) [reply spec/models/journey_spec.rb]: done

File: spec/models/journey_spec.rb (hunk 3 above)
Comment (muhammadanas123) [reply spec/models/journey_spec.rb]: done

**Pull request merged**